cmd/minikube/cmd/config/config_test.go

```diff
@@ -41,12 +41,14 @@ var configTestCases = []configTestCase{
     "ReminderWaitPeriodInHours": 99,
     "cpus": 4,
     "disk-size": "20g",
+    "iso-url": "http://foo.bar/minishift-centos.iso",
     "log_dir": "/etc/hosts",
     "show-libmachine-logs": true,
     "v": 5,
     "vm-driver": "kvm"
 }`,
 		config: map[string]interface{}{
+			"iso-url": "http://foo.bar/minishift-centos.iso",
```

Comment:
> Yes if you add manually then Viper will pick it up that's how viper works, it just read that config file

Exactly, so the minishift get/set is just something on top to allow the validation and setting of a specific set of flags. TBH, this bugs me a bit, since it is not consistent in behavior and it prevents me to set other flags which are not listed there. That's also why we initially struggled with understanding why Viper config file was not working as expected.

> I would say we should use a different config file and add some intelligent there to verify key-value pair which and that config can be modified by manually. 

I don't think another config file is needed. I think we can nicely solve this with a single file as well, but we can cross this bridge when we have to

> Currently config.json file used by viper and I am not sure if we can add some magic there to validate what key and respective value

We don't have to. However, if we wanted to we could.

> and that's why set and get make more sense here to modify that file instead manually where validator functions are already in place.

The get/set approach is what we are going to document for sure. 

One thing I don't like at all is that this _config set/get_ approach is not self documenting. One has no idea which values are supported. Either view should show the full MinikubeConfig applying defaults where appropriate or we need another command or something in the help which lists the available flags